### PR TITLE
24 backend api

### PR DESCRIPTION
**Pull request by paul-caron**

### CODE REVIEW

File: backend/src/controllers/codeController.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+export const createCode = errorHandler(async (req: AuthRequest, res: Response) => {
+    createCodeSchema.parse(req.body);
+    const language: CodeLanguage = req.body.language ;
+    const title: string = req.body.title ;
+    const source: string = req.body.source ;
+    const data = {
+        codeLanguage: language, title, source, userId:req.user!.id
+    }
+    try{
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:28]: remove all try catch from controllers, if you want throw certain http exception, throw it directly, that is why we wrap all with errorHandler

File: backend/src/controllers/codeController.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        res.json({ success: !!code.id });
+    }catch(e){
+        throw new ForbiddenException("Could not create code", ErrorCode.FORBIDDEN);
+    }
+});
+
+export const deleteCode = errorHandler(async (req: AuthRequest, res: Response) => {
+    deleteCodeSchema.parse(req.body);
+    const codeId: bigint = req.body.codeId ;
+    try{
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:39]: remove try/catch

File: backend/src/controllers/codeController.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    createCodeSchema.parse(req.body);
+    const language: CodeLanguage = req.body.language ;
+    const title: string = req.body.title ;
+    const source: string = req.body.source ;
+    const data = {
+        codeLanguage: language, title, source, userId:req.user!.id
+    }
+    try{
+        const code: Code = await dbClient.code.create({data});
+        res.json({ success: !!code.id });
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:30]: this is funny but use success: true

File: backend/src/controllers/codeController.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+export const updateCode = errorHandler(async (req: AuthRequest, res: Response) => {
+    updateCodeSchema.parse(req.body);
+    const codeId: bigint = req.body.codeId ;
+    const title: string = req.body.title ;
+    const source: string = req.body.source ;
+    const queryData = {
+        where:{
+            id: codeId,
+            userId:req.user!.id
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:55]: remove

File: backend/src/controllers/codeController.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        },
+        data: {
+            title,
+            source,
+        }
+    }
+    try{
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:62]: remove try/catch

File: backend/src/controllers/codeController.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+export const getCode = errorHandler(async (req: AuthRequest, res: Response) => {
+    getCodeSchema.parse(req.body);
+    const codeId: bigint = req.body.codeId ;
+    const queryData = {
+        where: {
+           id: codeId
+        }
+    };
+    try{
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:78]: remove try/catch

File: backend/src/controllers/codeController.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        await dbClient.code.delete({where:{id:codeId, userId: req.user!.id}});
+        res.json({ success: true });
+    }catch(e){
+        throw new ForbiddenException("Could not delete. Forbidden.", ErrorCode.FORBIDDEN);
+    }
+});
+
+export const updateCode = errorHandler(async (req: AuthRequest, res: Response) => {
+    updateCodeSchema.parse(req.body);
+    const codeId: bigint = req.body.codeId ;
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:49]: BigInt(req.body.codeId)

File: backend/src/controllers/codeController.ts [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        const code: Code = await dbClient.code.create({data});
+        res.json({ success: !!code.id });
+    }catch(e){
+        throw new ForbiddenException("Could not create code", ErrorCode.FORBIDDEN);
+    }
+});
+
+export const deleteCode = errorHandler(async (req: AuthRequest, res: Response) => {
+    deleteCodeSchema.parse(req.body);
+    const codeId: bigint = req.body.codeId ;
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:38]: wrap into BigInt()

File: backend/src/controllers/codeController.ts [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    try{
+        await dbClient.code.delete({where:{id:codeId, userId: req.user!.id}});
+        res.json({ success: true });
+    }catch(e){
+        throw new ForbiddenException("Could not delete. Forbidden.", ErrorCode.FORBIDDEN);
+    }
+});
+
+export const updateCode = errorHandler(async (req: AuthRequest, res: Response) => {
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:47]: missing `isPublic` property

File: backend/src/controllers/codeController.ts [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        const code: Code = await dbClient.code.update(queryData);
+        res.json({ success: !!code.id });
+    }catch(e){
+        throw new ForbiddenException("Could not update. Forbidden.", ErrorCode.FORBIDDEN);
+    }
+});
+
+export const getCode = errorHandler(async (req: AuthRequest, res: Response) => {
+    getCodeSchema.parse(req.body);
+    const codeId: bigint = req.body.codeId ;
```
Comment (DonDejvo) [inline backend/src/controllers/codeController.ts:72]: wrap BigInt()

File: backend/src/schemas/codeSchemas.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
+export const deleteCodeSchema = z.object({
+    codeId: idSchema,
+});
+
+export const getCodeSchema = z.object({
+    codeId: idSchema,
+});
+
+export const updateCodeSchema = z.object({
```
Comment (DonDejvo) [inline backend/src/schemas/codeSchemas.ts:17]: title, source, isPublic all should be optional, this way is better performance

File: backend/src/controllers/codeController.ts (hunk 7 above)
Comment (paul-caron) [reply backend/src/controllers/codeController.ts:49]: bigint no longer required, as per discussion

File: backend/src/controllers/codeController.ts (hunk 8 above)
Comment (paul-caron) [reply backend/src/controllers/codeController.ts:38]: bigint no longer required

File: backend/src/controllers/codeController.ts (hunk 10 above)
Comment (paul-caron) [reply backend/src/controllers/codeController.ts:72]: bigint no longer required

File: backend/src/controllers/codeController.ts (hunk 6 above)
Comment (paul-caron) [reply backend/src/controllers/codeController.ts:78]: removed

File: backend/src/controllers/codeController.ts (hunk 5 above)
Comment (paul-caron) [reply backend/src/controllers/codeController.ts:62]: removed

File: backend/src/controllers/codeController.ts (hunk 3 above)
Comment (paul-caron) [reply backend/src/controllers/codeController.ts:30]: done

File: backend/src/controllers/codeController.ts (hunk 2 above)
Comment (paul-caron) [reply backend/src/controllers/codeController.ts:39]: removed

File: backend/src/controllers/codeController.ts (hunk 1 above)
Comment (paul-caron) [reply backend/src/controllers/codeController.ts:28]: removed all the try catch

File: backend/src/controllers/codeController.ts (hunk 9 above)
Comment (paul-caron) [reply backend/src/controllers/codeController.ts:40]: done. isPublic property added.

File: backend/src/schemas/codeSchemas.ts (hunk 11 above)
Comment (paul-caron) [reply backend/src/schemas/codeSchemas.ts:17]: done. they are optional now

**Pull request merged**